File: camerax_android/lib/src/legacy/core/mirror_mode.dart
import 'package:camerax_android/src/legacy/camerax.g.dart' as $native;
import 'package:camerax_platform_interface/camerax_platform_interface.dart'
    as $interface;

extension MirrorModeArgs on $interface.MirrorMode {
  $native.MirrorMode get obj {
    switch (this) {
      case $interface.MirrorMode.off:
        return $native.MirrorMode.off;
      case $interface.MirrorMode.on:
        return $native.MirrorMode.on;
      case $interface.MirrorMode.onFrontOnly:
        return $native.MirrorMode.onFrontOnly;
    }
  }
}

extension MirrorModeObj on $native.MirrorMode {
  $interface.MirrorMode get args {
    switch (this) {
      case $native.MirrorMode.off:
        return $interface.MirrorMode.off;
      case $native.MirrorMode.on:
        return $interface.MirrorMode.on;
      case $native.MirrorMode.onFrontOnly:
        return $interface.MirrorMode.onFrontOnly;
    }
  }
}

File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/legacy/core/MirrorMode.kt
package dev.hebei.camerax_android.legacy.core

import dev.hebei.camerax_android.legacy.MirrorMode

val MirrorMode.obj
    get() = when (this) {
        MirrorMode.OFF -> androidx.camera.core.MirrorMode.MIRROR_MODE_OFF
        MirrorMode.ON -> androidx.camera.core.MirrorMode.MIRROR_MODE_ON
        MirrorMode.ON_FRONT_ONLY -> androidx.camera.core.MirrorMode.MIRROR_MODE_ON_FRONT_ONLY
    }

val Int.mirrorModeArgs
    get() = when (this) {
        androidx.camera.core.MirrorMode.MIRROR_MODE_OFF -> MirrorMode.OFF
        androidx.camera.core.MirrorMode.MIRROR_MODE_ON -> MirrorMode.ON
        androidx.camera.core.MirrorMode.MIRROR_MODE_ON_FRONT_ONLY -> MirrorMode.ON_FRONT_ONLY
        else -> throw IllegalArgumentException()
    }